### PR TITLE
feat: add table collection

### PR DESCRIPTION
**Pull request by martinmaillard**

### CODE REVIEW

File: ui/src/store/modules/cube-projects.ts [hunk 1]
```diff
@@ -2,23 +2,32 @@ import { ActionTree, MutationTree, GetterTree } from 'vuex'
...
 import * as ns from '@cube-creator/core/namespace'
-import { Project, ProjectsCollection, CSVMapping, SourcesCollection } from '@/types'
+import { Project, ProjectsCollection, CSVMapping, SourcesCollection, TableCollection } from '@/types'
 
 export interface CubeProjectsState {
   collection: null | ProjectsCollection,
   project: null | Project,
   csvMapping: null | CSVMapping,
   sourcesCollection: null | SourcesCollection,
+  tableCollection: null | TableCollection,
```
Comment (tpluscode) [inline ui/src/store/modules/cube-projects.ts:12]: Ditto, maybe we could drop the `Collection` suffix in property names

File: ui/src/api/mixins/TableCollection.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return class extends base implements TableCollection {
+    get actions () {
+      return {
+        create: findOperation(this, schema.CreateAction),
+      }
+    }
+  }
+}
+
+Mixin.appliesTo = ns.cc.TableCollection
```
Comment (tpluscode) [inline ui/src/api/mixins/TableCollection.ts:19]: Not to change this now but wonder if you really need this as a mixin. Food for thought if we indeed use `schema` for common actions

File: ui/src/api/mixins/Table.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Resource } from 'alcaeus'
+import * as ns from '@cube-creator/core/namespace'
+import { schema } from '@tpluscode/rdf-ns-builders'
+import { Table, Source } from '@/types'
+import { findOperation } from '../common'
+
+export type Constructor = new (...args: any[]) => Resource;
+
+export default function Mixin<Base extends Constructor> (base: Base) {
```
Comment (tpluscode) [inline ui/src/api/mixins/Table.ts:9]: I see what you did there 🙂. The patterns goes like this

```suggestion
import { Constructor } from '@tpluscode/rdfine'

export default function Mixin<Base extends Constructor<Resource> (base: Base) {
```

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: ui/src/types.ts [hunk 4]
```diff
@@ -12,9 +12,16 @@ export interface SourcesCollection extends Collection {
...
+  actions: {
+    create: RuntimeOperation | null,
+  };
+}
+
 export interface CSVMapping extends Resource {
   actions: Record<string, unknown>;
-  sourcesCollection: SourcesCollection,
+  sourcesCollection: SourcesCollection;
+  tableCollection: TableCollection;
```
[the remark this answers is not in the document]
Comment (martinmaillard) [reply ui/src/types.ts:24]: I prefer to keep `tables` for an actual list of tables in the frontend. It's easier to understand.

File: ui/src/api/mixins/TableCollection.ts (hunk 2 above)
Comment (martinmaillard) [reply ui/src/api/mixins/TableCollection.ts:19]: Yeah I'm not doing anything else that extracting the operations in `actions` at the moment so if we use schema for all of the operations, I can have a generic mixin for collections.

File: ui/src/api/mixins/Table.ts (hunk 3 above)
Comment (martinmaillard) [reply ui/src/api/mixins/Table.ts:9]: Alright, will change 👍 
I guess it was a pattern I ported from the first project.

File: apis/core/lib/domain/cube-projects/create.ts [hunk 5]
```diff
@@ -50,7 +50,25 @@ export async function createProject({
...
+      .addOut(hydra.manages, manages => {
+        // ?member rdf:type cc:Table
+        manages.addOut([hydra.property], [rdf.type])
+        manages.addOut([hydra.object], [cc.Table])
+      })
+      .addOut(hydra.manages, manages => {
+        // ?member cc:csvMapping <mapping>
+        manages.addOut([hydra.object], mapping)
+        manages.addOut([hydra.property], [cc.csvSource])
+      })
```
[the remark this answers is not in the document]
Comment (martinmaillard) [reply apis/core/lib/domain/cube-projects/create.ts:69]: Actually I think the comment was correct, but the `cc:csvSource` property should have been `cc:csvMapping`

File: ui/src/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tpluscode) [reply ui/src/types.ts:24]: Fair enough

**Review by tpluscode — APPROVED** [review]

**Pull request merged**